CPTextView/Frameworks/TextView/CPTypesetter.j

```diff
@@ -399,22 +427,24 @@ var CPSystemTypesetterFactory;
 
             if (isNewline)
             {
-                if ([_currentParagraph minimumLineHeight])
-                    _lineHeight = MAX(_lineHeight, [_currentParagraph minimumLineHeight]);
+                if (currentParagraphMinimumLineHeight && currentParagraphMinimumLineHeight > _lineHeight)
+                    _lineHeight = currentParagraphMinimumLineHeight;
 
-                if ([_currentParagraph maximumLineHeight])
-                    _lineHeight = MIN(_lineHeight, [_currentParagraph maximumLineHeight]);
+                if (currentParagraphMaximumLineHeight && currentParagraphMaximumLineHeight < _lineHeight)
+                    _lineHeight = currentParagraphMaximumLineHeight;
 
                 lineOrigin.y += _lineHeight;
 
-                if ([_currentParagraph lineSpacing])
-                    lineOrigin.y += [_currentParagraph lineSpacing];
+                if (currentParagraphLineSpacing)
+                    lineOrigin.y += currentParagraphLineSpacing;
 
-                if (lineOrigin.y > [_currentTextContainer containerSize].height)
+                if (lineOrigin.y > containerSizeHeight && _indexOfCurrentContainer < textContainersCount - 1)
```

Comment:
this should happen very rarely. i do not think that [[_layoutManager textContainers] count] is a bottleneck